diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdf9f99..29469fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 3.16)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 
 project(KaMinPar CXX C)
-set(PROJECT_VENDOR "Daniel Seemaier")
-set(PROJECT_CONTACT "daniel.seemaier@kit.edu")
+set(PROJECT_VENDOR "XXX")
+set(PROJECT_CONTACT "XXX")
 
 set(CMAKE_CXX_STANDARD 20)
 
@@ -17,6 +17,7 @@ option(KAMINPAR_ENABLE_TIMERS "Enable fine-grained timers. With this option disa
 option(KAMINPAR_ENABLE_DEBUG_FEATURES "Enable features that might slow down execution but are helpful for debug purposes (e.g., naming graphs)." OFF)
 option(KAMINPAR_ENABLE_STATISTICS "Generate and output much more detailed statistics." OFF)
 option(KAMINPAR_64BIT_EDGE_IDS "Use 64 bits for edge ids." OFF)
+option(DKAMINPAR_BUILD_METIS_INTERFACE "Generate library in the metis format" ON)
 
 include_directories(${PROJECT_SOURCE_DIR})
 
@@ -111,4 +112,4 @@ include(gmock)
 add_subdirectory(kaminpar)
 add_subdirectory(apps)
 add_subdirectory(tests)
-add_subdirectory(library)
\ No newline at end of file
+add_subdirectory(library)
