diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdf9f99..29469fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 3.16)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
 
 project(KaMinPar CXX C)
-set(PROJECT_VENDOR "Daniel Seemaier")
-set(PROJECT_CONTACT "daniel.seemaier@kit.edu")
+set(PROJECT_VENDOR "XXX")
+set(PROJECT_CONTACT "XXX")
 
 set(CMAKE_CXX_STANDARD 20)
 
@@ -17,6 +17,7 @@ option(KAMINPAR_ENABLE_TIMERS "Enable fine-grained timers. With this option disa
 option(KAMINPAR_ENABLE_DEBUG_FEATURES "Enable features that might slow down execution but are helpful for debug purposes (e.g., naming graphs)." OFF)
 option(KAMINPAR_ENABLE_STATISTICS "Generate and output much more detailed statistics." OFF)
 option(KAMINPAR_64BIT_EDGE_IDS "Use 64 bits for edge ids." OFF)
+option(DKAMINPAR_BUILD_METIS_INTERFACE "Generate library in the metis format" ON)
 
 include_directories(${PROJECT_SOURCE_DIR})
 
@@ -111,4 +112,4 @@ include(gmock)
 add_subdirectory(kaminpar)
 add_subdirectory(apps)
 add_subdirectory(tests)
-add_subdirectory(library)
\ No newline at end of file
+add_subdirectory(library)
diff --git a/README.MD b/README.MD
index 65264f7..67dca47 100644
--- a/README.MD
+++ b/README.MD
@@ -79,26 +79,3 @@ Then, simply link against our library (together with `libtbb` and `libtbbmalloc`
 ## Licensing
 
 KaMinPar is free software provided under the GNU General Public License (GPLv3).
-If you publish results using our algorithms, please acknowledge our work by citing the following paper ([pdf](http://algo2.iti.kit.edu/seemaier/deep_mgp/)):
-
-```
-@inproceedings{DBLP:conf/esa/GottesburenH00S21,
-  author    = {Lars Gottesb{\"{u}}ren and
-               Tobias Heuer and
-               Peter Sanders and
-               Christian Schulz and
-               Daniel Seemaier},
-  title     = {Deep Multilevel Graph Partitioning},
-  booktitle = {29th Annual European Symposium on Algorithms, {ESA} 2021, September
-               6-8, 2021, Lisbon, Portugal (Virtual Conference)},
-  series    = {LIPIcs},
-  volume    = {204},
-  pages     = {48:1--48:17},
-  publisher = {Schloss Dagstuhl - Leibniz-Zentrum f{\"{u}}r Informatik},
-  year      = {2021},
-  url       = {https://doi.org/10.4230/LIPIcs.ESA.2021.48},
-  doi       = {10.4230/LIPIcs.ESA.2021.48}
-}
-```
-
-Detailed experimental results published in our paper are available [here](http://algo2.iti.kit.edu/seemaier/deep_mgp/).
\ No newline at end of file
diff --git a/apps/apps.h b/apps/apps.h
index 18fea97..4655891 100644
--- a/apps/apps.h
+++ b/apps/apps.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/apps/tools/graph_statistics.cc b/apps/tools/graph_statistics.cc
index 7594a71..bc58966 100644
--- a/apps/tools/graph_statistics.cc
+++ b/apps/tools/graph_statistics.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/algorithm/extract_subgraphs.cc b/kaminpar/algorithm/extract_subgraphs.cc
index fa491c7..76c0aee 100644
--- a/kaminpar/algorithm/extract_subgraphs.cc
+++ b/kaminpar/algorithm/extract_subgraphs.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -250,4 +250,4 @@ SubgraphExtractionResult extract_subgraphs(const PartitionedGraph &p_graph, Subg
 
   return {std::move(subgraphs), std::move(mapping), std::move(start_positions)};
 }
-} // namespace kaminpar::graph
\ No newline at end of file
+} // namespace kaminpar::graph
diff --git a/kaminpar/algorithm/extract_subgraphs.h b/kaminpar/algorithm/extract_subgraphs.h
index 4bb459c..6073c5d 100644
--- a/kaminpar/algorithm/extract_subgraphs.h
+++ b/kaminpar/algorithm/extract_subgraphs.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -119,4 +119,4 @@ SequentialSubgraphExtractionResult extract_subgraphs_sequential(const Partitione
                                                                 SubgraphMemoryStartPosition memory_position,
                                                                 SubgraphMemory &subgraph_memory,
                                                                 TemporarySubgraphMemory &tmp_subgraph_memory);
-} // namespace kaminpar::graph
\ No newline at end of file
+} // namespace kaminpar::graph
diff --git a/kaminpar/algorithm/graph_contraction.cc b/kaminpar/algorithm/graph_contraction.cc
index 2265d84..fafbd41 100644
--- a/kaminpar/algorithm/graph_contraction.cc
+++ b/kaminpar/algorithm/graph_contraction.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -223,4 +223,4 @@ ContractionResult contract(const Graph &graph, const scalable_vector<NodeID> &cl
           std::move(mapping), std::move(m_ctx)};
 }
 
-} // namespace kaminpar::graph
\ No newline at end of file
+} // namespace kaminpar::graph
diff --git a/kaminpar/algorithm/graph_contraction.h b/kaminpar/algorithm/graph_contraction.h
index d245c22..36c544f 100644
--- a/kaminpar/algorithm/graph_contraction.h
+++ b/kaminpar/algorithm/graph_contraction.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -85,4 +85,4 @@ struct ContractionResult {
 
 ContractionResult contract(const Graph &r, const scalable_vector<NodeID> &clustering,
                            ContractionMemoryContext m_ctx = {});
-} // namespace kaminpar::graph
\ No newline at end of file
+} // namespace kaminpar::graph
diff --git a/kaminpar/algorithm/graph_utils.h b/kaminpar/algorithm/graph_utils.h
index 766ffbf..5c39df1 100644
--- a/kaminpar/algorithm/graph_utils.h
+++ b/kaminpar/algorithm/graph_utils.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/algorithm/parallel_label_propagation.h b/kaminpar/algorithm/parallel_label_propagation.h
index 80d6dc9..59a9891 100644
--- a/kaminpar/algorithm/parallel_label_propagation.h
+++ b/kaminpar/algorithm/parallel_label_propagation.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/application/arguments.h b/kaminpar/application/arguments.h
index 4895eff..8d753c7 100644
--- a/kaminpar/application/arguments.h
+++ b/kaminpar/application/arguments.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -183,4 +183,4 @@ Context parse_options(int argc, char *argv[]) {
   arguments.parse(argc, argv);
   return context;
 }
-} // namespace kaminpar::app
\ No newline at end of file
+} // namespace kaminpar::app
diff --git a/kaminpar/coarsening/parallel_label_propagation_coarsener.h b/kaminpar/coarsening/parallel_label_propagation_coarsener.h
index 15e3cb6..f69924a 100644
--- a/kaminpar/coarsening/parallel_label_propagation_coarsener.h
+++ b/kaminpar/coarsening/parallel_label_propagation_coarsener.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/datastructure/rating_map.h b/kaminpar/datastructure/rating_map.h
index 98bb750..88ef28a 100644
--- a/kaminpar/datastructure/rating_map.h
+++ b/kaminpar/datastructure/rating_map.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -79,4 +79,4 @@ private:
   std::size_t _super_small_map_counter{0};
   std::size_t _large_map_counter{0};
 };
-} // namespace kaminpar
\ No newline at end of file
+} // namespace kaminpar
diff --git a/kaminpar/initial_partitioning/initial_coarsener.cc b/kaminpar/initial_partitioning/initial_coarsener.cc
index fa21cb7..6bfb1f2 100644
--- a/kaminpar/initial_partitioning/initial_coarsener.cc
+++ b/kaminpar/initial_partitioning/initial_coarsener.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/initial_partitioning/initial_coarsener.h b/kaminpar/initial_partitioning/initial_coarsener.h
index 43d2d16..cfe8826 100644
--- a/kaminpar/initial_partitioning/initial_coarsener.h
+++ b/kaminpar/initial_partitioning/initial_coarsener.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -178,4 +178,4 @@ private:
   Randomize &_rand{Randomize::instance()};
   static RandomPermutations<NodeID, kChunkSize, kNumberOfNodePermutations> _random_permutations;
 };
-} // namespace kaminpar::ip
\ No newline at end of file
+} // namespace kaminpar::ip
diff --git a/kaminpar/initial_partitioning/initial_partitioning_facade.h b/kaminpar/initial_partitioning/initial_partitioning_facade.h
index f02e011..a496333 100644
--- a/kaminpar/initial_partitioning/initial_partitioning_facade.h
+++ b/kaminpar/initial_partitioning/initial_partitioning_facade.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -140,4 +140,4 @@ private:
   BlockID _final_k2;
   std::size_t _num_bipartition_repetitions;
 };
-} // namespace kaminpar::ip
\ No newline at end of file
+} // namespace kaminpar::ip
diff --git a/kaminpar/initial_partitioning/initial_refiner.h b/kaminpar/initial_partitioning/initial_refiner.h
index 93b7097..0c6cabd 100644
--- a/kaminpar/initial_partitioning/initial_refiner.h
+++ b/kaminpar/initial_partitioning/initial_refiner.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -443,4 +443,4 @@ using InitialSimple2WayFM = InitialTwoWayFMRefiner<fm::MaxOverloadSelectionPolic
                                                    fm::SimpleStoppingPolicy>;
 using InitialAdaptive2WayFM = InitialTwoWayFMRefiner<fm::MaxOverloadSelectionPolicy, fm::BalancedMinCutAcceptancePolicy,
                                                      fm::AdaptiveStoppingPolicy>;
-} // namespace kaminpar::ip
\ No newline at end of file
+} // namespace kaminpar::ip
diff --git a/kaminpar/initial_partitioning/sequential_graph_hierarchy.cc b/kaminpar/initial_partitioning/sequential_graph_hierarchy.cc
index 5d58c85..6d81785 100644
--- a/kaminpar/initial_partitioning/sequential_graph_hierarchy.cc
+++ b/kaminpar/initial_partitioning/sequential_graph_hierarchy.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/initial_partitioning/sequential_graph_hierarchy.h b/kaminpar/initial_partitioning/sequential_graph_hierarchy.h
index c7e1de3..b49aaf5 100644
--- a/kaminpar/initial_partitioning/sequential_graph_hierarchy.h
+++ b/kaminpar/initial_partitioning/sequential_graph_hierarchy.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -50,4 +50,4 @@ private:
   std::vector<std::vector<NodeID>> _coarse_mappings;
   std::vector<Graph> _coarse_graphs;
 };
-} // namespace kaminpar::ip
\ No newline at end of file
+} // namespace kaminpar::ip
diff --git a/kaminpar/partitioning_scheme/helper.h b/kaminpar/partitioning_scheme/helper.h
index da28a36..deca9a9 100644
--- a/kaminpar/partitioning_scheme/helper.h
+++ b/kaminpar/partitioning_scheme/helper.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -129,4 +129,4 @@ inline bool parallel_ip_mode(const InitialPartitioningMode &mode) {
          mode == InitialPartitioningMode::SYNCHRONOUS_PARALLEL;
 }
 } // namespace helper
-} // namespace kaminpar::partitioning
\ No newline at end of file
+} // namespace kaminpar::partitioning
diff --git a/kaminpar/partitioning_scheme/parallel_initial_partitioner.cc b/kaminpar/partitioning_scheme/parallel_initial_partitioner.cc
index 1ce9f42..6c563ed 100644
--- a/kaminpar/partitioning_scheme/parallel_initial_partitioner.cc
+++ b/kaminpar/partitioning_scheme/parallel_initial_partitioner.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -86,4 +86,4 @@ PartitionedGraph ParallelInitialPartitioner::split_and_join(const Coarsener *coa
   const std::size_t best = helper::select_best(p_graphs, p_ctx);
   return std::move(p_graphs[best]);
 }
-} // namespace kaminpar::partitioning
\ No newline at end of file
+} // namespace kaminpar::partitioning
diff --git a/kaminpar/partitioning_scheme/parallel_initial_partitioner.h b/kaminpar/partitioning_scheme/parallel_initial_partitioner.h
index 98b0eb8..07d111b 100644
--- a/kaminpar/partitioning_scheme/parallel_initial_partitioner.h
+++ b/kaminpar/partitioning_scheme/parallel_initial_partitioner.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/partitioning_scheme/parallel_recursive_bisection.cc b/kaminpar/partitioning_scheme/parallel_recursive_bisection.cc
index 16389cb..f984899 100644
--- a/kaminpar/partitioning_scheme/parallel_recursive_bisection.cc
+++ b/kaminpar/partitioning_scheme/parallel_recursive_bisection.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/partitioning_scheme/parallel_recursive_bisection.h b/kaminpar/partitioning_scheme/parallel_recursive_bisection.h
index dcabd98..2870883 100644
--- a/kaminpar/partitioning_scheme/parallel_recursive_bisection.h
+++ b/kaminpar/partitioning_scheme/parallel_recursive_bisection.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -88,4 +88,4 @@ private:
   // Initial partitioning
   GlobalInitialPartitionerMemoryPool _ip_m_ctx_pool;
 };
-} // namespace kaminpar::partitioning
\ No newline at end of file
+} // namespace kaminpar::partitioning
diff --git a/kaminpar/partitioning_scheme/parallel_simple_recursive_bisection.h b/kaminpar/partitioning_scheme/parallel_simple_recursive_bisection.h
index fe33dcf..af41df6 100644
--- a/kaminpar/partitioning_scheme/parallel_simple_recursive_bisection.h
+++ b/kaminpar/partitioning_scheme/parallel_simple_recursive_bisection.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -121,4 +121,4 @@ private:
 
   GlobalInitialPartitionerMemoryPool ip_m_ctx_pool;
 };
-} // namespace kaminpar::partitioning
\ No newline at end of file
+} // namespace kaminpar::partitioning
diff --git a/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.cc b/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.cc
index 72b427c..7aabf47 100644
--- a/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.cc
+++ b/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.h b/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.h
index 8850261..dd44efa 100644
--- a/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.h
+++ b/kaminpar/partitioning_scheme/parallel_synchronized_initial_partitioner.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/partitioning_scheme/partitioning.h b/kaminpar/partitioning_scheme/partitioning.h
index c47b1fe..c2169d3 100644
--- a/kaminpar/partitioning_scheme/partitioning.h
+++ b/kaminpar/partitioning_scheme/partitioning.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -24,4 +24,4 @@
 
 namespace kaminpar::partitioning {
 PartitionedGraph partition(const Graph &graph, const Context &ctx);
-}
\ No newline at end of file
+}
diff --git a/kaminpar/refinement/parallel_balancer.cc b/kaminpar/refinement/parallel_balancer.cc
index fc2f324..8bb5f72 100644
--- a/kaminpar/refinement/parallel_balancer.cc
+++ b/kaminpar/refinement/parallel_balancer.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -310,4 +310,4 @@ void ParallelBalancer::init_feasible_target_blocks() {
     if (_p_graph->block_weight(b) < _p_ctx->perfectly_balanced_block_weight(b)) { blocks.push_back(b); }
   }
 }
-} // namespace kaminpar
\ No newline at end of file
+} // namespace kaminpar
diff --git a/kaminpar/refinement/parallel_balancer.h b/kaminpar/refinement/parallel_balancer.h
index 6f8d764..4a32a0d 100644
--- a/kaminpar/refinement/parallel_balancer.h
+++ b/kaminpar/refinement/parallel_balancer.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/refinement/parallel_label_propagation_refiner.h b/kaminpar/refinement/parallel_label_propagation_refiner.h
index 4b35a17..fe1cbfd 100644
--- a/kaminpar/refinement/parallel_label_propagation_refiner.h
+++ b/kaminpar/refinement/parallel_label_propagation_refiner.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/kaminpar/utility/timer.cc b/kaminpar/utility/timer.cc
index d9bb279..9b03e41 100644
--- a/kaminpar/utility/timer.cc
+++ b/kaminpar/utility/timer.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -156,4 +156,4 @@ timer::FlatTimer &timer::FlatTimer::global() {
   static FlatTimer instance{};
   return instance;
 }
-} // namespace kaminpar
\ No newline at end of file
+} // namespace kaminpar
diff --git a/kaminpar/utility/timer.h b/kaminpar/utility/timer.h
index f09717d..c188069 100644
--- a/kaminpar/utility/timer.h
+++ b/kaminpar/utility/timer.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
diff --git a/library/example/library_example_from_file.cc b/library/example/library_example_from_file.cc
index a51eed3..829ada3 100644
--- a/library/example/library_example_from_file.cc
+++ b/library/example/library_example_from_file.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -48,4 +48,4 @@ int main(int argc, char *argv[]) {
   for (int b = 0; b < k; ++b) { std::cout << " block " << b << ": " << block_sizes[b] << std::endl; }
 
   return 0;
-}
\ No newline at end of file
+}
diff --git a/library/example/library_example_static.cc b/library/example/library_example_static.cc
index 494b2b7..a0d53db 100644
--- a/library/example/library_example_static.cc
+++ b/library/example/library_example_static.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -45,4 +45,4 @@ int main(int, char *[]) {
   for (int b = 0; b < k; ++b) { std::cout << " block " << b << ": " << block_sizes[b] << std::endl; }
 
   return 0;
-}
\ No newline at end of file
+}
diff --git a/library/libkaminpar.cc b/library/libkaminpar.cc
index 25708c5..0c45b3b 100644
--- a/library/libkaminpar.cc
+++ b/library/libkaminpar.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -220,4 +220,4 @@ std::unique_ptr<BlockID[]> Partitioner::partition(BlockID k) const {
 }
 
 std::size_t Partitioner::partition_size() const { return static_cast<std::size_t>(_pimpl->n); }
-} // namespace libkaminpar
\ No newline at end of file
+} // namespace libkaminpar
diff --git a/library/libkaminpar.h b/library/libkaminpar.h
index e3226bc..ea638e2 100644
--- a/library/libkaminpar.h
+++ b/library/libkaminpar.h
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -72,4 +72,4 @@ private:
 private:
   struct PartitionerBuilderPrivate *_pimpl;
 };
-} // namespace libkaminpar
\ No newline at end of file
+} // namespace libkaminpar
diff --git a/library/metis_adapter.cc b/library/metis_adapter.cc
index 0d52783..7264959 100644
--- a/library/metis_adapter.cc
+++ b/library/metis_adapter.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2021 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2021 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -161,4 +161,4 @@ METIS_ComputeVertexSeparator(idx_t *, idx_t *, idx_t *, idx_t *, idx_t *, idx_t
 
 METIS_API(int)
 METIS_NodeRefine(idx_t, idx_t *, idx_t *, idx_t *, idx_t *, idx_t *, real_t) { return METIS_ERROR; }
-}
\ No newline at end of file
+}
diff --git a/tests/datastructure/graph_test.cc b/tests/datastructure/graph_test.cc
index 9f3476a..e004ef8 100644
--- a/tests/datastructure/graph_test.cc
+++ b/tests/datastructure/graph_test.cc
@@ -1,7 +1,7 @@
 /*******************************************************************************
  * This file is part of KaMinPar.
  *
- * Copyright (C) 2020 Daniel Seemaier <daniel.seemaier@kit.edu>
+ * Copyright (C) 2020 XXX<XXX>
  *
  * KaMinPar is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -231,4 +231,4 @@ TEST(GraphTest, LowestDegreeInBucketWorks) {
   EXPECT_EQ(lowest_degree_in_bucket(2), 2);
   EXPECT_EQ(lowest_degree_in_bucket(3), 4);
 }
-} // namespace kaminpar
\ No newline at end of file
+} // namespace kaminpar
